plugins/woocommerce/src/Blocks/BlockTemplatesController.php

```diff
@@ -43,7 +46,37 @@ public function render_woocommerce_template_part( $attributes ) {
 			$template_part = get_block_template( $attributes['theme'] . '//' . $attributes['slug'], 'wp_template_part' );
 
 			if ( $template_part && ! empty( $template_part->content ) ) {
-				return do_blocks( $template_part->content );
+				$content            = do_blocks( $template_part->content );
```

Comment:
We were missing a wrapper here. The core function outputs a wrapper (header in this case) but since the templates come from the plugin we need to take care of it.